packages/x-date-pickers/src/internals/components/PickerFieldUI.tsx

```diff
@@ -0,0 +1,501 @@
+      'aria-label': openPickerAriaLabel,
+      edge:
+        clearButtonPosition === 'start' && openPickerButtonPosition === 'start'
+          ? undefined
+          : openPickerButtonPosition,
+    },
+    ownerState,
+  });
+
+  const OpenPickerIcon = pickerFieldUIContext.slots.openPickerIcon ?? defaultOpenPickerIcon;
+  const openPickerIconProps = useSlotProps({
+    elementType: OpenPickerIcon,
+    externalSlotProps: pickerFieldUIContext.slotProps.openPickerIcon,
+    ownerState,
+  });
+
+  const ClearButton = slots?.clearButton ?? pickerFieldUIContext.slots.clearButton ?? MuiIconButton;
+  // We don't want to forward the `ownerState` to the `<IconButton />` component, see mui/material-ui#34056
+  const { ownerState: clearButtonOwnerState, ...clearButtonProps } = useSlotProps({
+    elementType: ClearButton,
+    externalSlotProps: mergeSlotProps(
+      pickerFieldUIContext.slotProps.clearButton,
+      slotProps?.clearButton,
+    ),
+    className: 'clearButton',
+    additionalProps: {
+      title: translations.fieldClearLabel,
+      tabIndex: -1,
+      onClick: onClear,
+      disabled: fieldResponse.disabled || fieldResponse.readOnly,
+      edge:
+        clearButtonPosition === 'end' && openPickerButtonPosition === 'end'
+          ? undefined
+          : clearButtonPosition,
+      'data-hidden': fieldResponse.areAllSectionsEmpty ? '' : undefined,
+    },
+    ownerState,
+  });
+
+  const ClearIcon = slots?.clearIcon ?? pickerFieldUIContext.slots.clearIcon ?? MuiClearIcon;
+  const clearIconProps = useSlotProps({
+    elementType: ClearIcon,
+    externalSlotProps: mergeSlotProps(
+      pickerFieldUIContext.slotProps.clearIcon,
+      slotProps?.clearIcon,
+    ),
+    additionalProps: {
+      fontSize: 'small',
+    },
+    ownerState,
+  });
+
+  if (!textFieldProps.InputProps) {
+    textFieldProps.InputProps = {};
+  }
+
+  if (pickerContext) {
+    textFieldProps.InputProps.ref = pickerContext.triggerRef;
+  }
+
+  if (
+    !textFieldProps.InputProps?.startAdornment &&
+    (clearButtonPosition === 'start' || openPickerButtonPosition === 'start')
+  ) {
+    textFieldProps.InputProps.startAdornment = (
+      <InputAdornment {...startInputAdornmentProps}>
+        {openPickerButtonPosition === 'start' && (
+          <OpenPickerButton {...openPickerButtonProps}>
+            <OpenPickerIcon {...openPickerIconProps} />
+          </OpenPickerButton>
+        )}
+        {clearButtonPosition === 'start' && (
+          <ClearButton {...clearButtonProps}>
+            <ClearIcon {...clearIconProps} />
+          </ClearButton>
+        )}
+      </InputAdornment>
+    );
+  }
+
+  if (
+    !textFieldProps.InputProps?.endAdornment &&
+    (clearButtonPosition === 'end' || openPickerButtonPosition === 'end')
+  ) {
+    textFieldProps.InputProps.endAdornment = (
+      <InputAdornment {...endInputAdornmentProps}>
+        {clearButtonPosition === 'end' && (
+          <ClearButton {...clearButtonProps}>
```

Comment:
Based on the test failures we can see that this is a behavior change.
For now, the clear button is only rendered when the field is not empty.
Do we want/need this change?

Reply:
No, I mis-read the previous version
Reverted